l1-contracts/src/core/Rollup.sol

```diff
@@ -31,9 +30,10 @@ import {Leonidas} from "./sequencer_selection/Leonidas.sol";
  * @notice Rollup contract that is concerned about readability and velocity of development
  * not giving a damn about gas costs.
  */
-contract Rollup is Leonidas, IRollup {
+contract Rollup is Leonidas, IRollup, ITestRollup {
   struct BlockLog {
     bytes32 archive;
+    uint128 slotNumber;
```

Comment:
They are not directly linked. Every slot is essentially an option for a block, so you don't really know what slot it belongs to if not explicit. You can say that it will implicitly be part of the archive if you open up the last header, but this was essentially as it was easier to provide the information and more clear what validation we were doing.